src/app/ngrx/reducers/podcast-totals.reducer.ts

```diff
@@ -1,59 +1,59 @@
 import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
-import { PodcastTotals, podcastTotalsKey } from './models';
+import { PodcastTotals, podcastTotalsId } from './models';
 import { ActionTypes, AllActions } from '../actions';
 
 export type State = EntityState<PodcastTotals>;
 
-export const adapter: EntityAdapter<PodcastTotals> = createEntityAdapter<PodcastTotals>({
-  selectId: ((p: PodcastTotals) => p.key)
-});
+export const adapter: EntityAdapter<PodcastTotals> = createEntityAdapter<PodcastTotals>();
 
 export const initialState: State = adapter.getInitialState({
   // additional entity state properties
 });
 
+export const {
+  selectIds,
+  selectEntities,
+  selectAll,
+} = adapter.getSelectors();
+
+export const selectPodcastTotalsIds = selectIds;
+export const selectPodcastTotalsEntities = selectEntities;
+export const selectAllPodcastTotals = selectAll;
+
 export function reducer(
   state = initialState,
   action: AllActions
 ): State {
   switch (action.type) {
     case ActionTypes.CASTLE_PODCAST_TOTALS_LOAD: {
       const { id, group, filter, beginDate, endDate } = action.payload;
-      const key = podcastTotalsKey(id, group, filter, beginDate, endDate);
-      return {
-        ...adapter.upsertOne({
+      const key = podcastTotalsId(id, group, filter, beginDate, endDate);
```

Comment:
TODO: I wanted to sort out this key vs id thing. Use podcastId (not id) when it is a combo key.